ktor-client/ktor-client-core/common/src/io/ktor/client/request/HttpRequest.kt

```diff
@@ -309,3 +309,11 @@ public fun HttpRequestData.isUpgradeRequest(): Boolean {
 public fun HttpRequestData.isSseRequest(): Boolean {
     return body is SSEClientContent
 }
+
+@InternalAPI
+@Suppress("KDocMissingDocumentation")
+public fun needToProcessSSE(data: HttpRequestData, status: HttpStatusCode, headers: Headers): Boolean {
+    return data.isSseRequest() &&
+        status == HttpStatusCode.OK &&
```

Comment:
not sure if it's needed here, but can we check that status is successful instead of OK?

Reply:
Can we leave `OK` because in spec it says next: `if res's status is not 200, or if res's Content-Type is not text/event-stream, then fail the connection`?